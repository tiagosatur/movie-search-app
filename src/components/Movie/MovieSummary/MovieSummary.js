import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faFileImage } from '@fortawesome/free-regular-svg-icons'
import styled, { css } from 'styled-components';
import { colorPallete, rem } from '../../../style'

const MovieSummary = props => {
    const {
        data: {
            Title,
            Plot,
            Poster
        },
    } = props;

    return (
        <StyledMovieSummary>
            <StyledPosterWrapper>
                {Poster !== 'N/A' 
                    ? <StyledSmallImage src={Poster} alt={Title} /> 
                    : <div>
                        <FontAwesomeIcon icon={faFileImage} style={{fontSize: '100px'}} />
                    </div>
                }
            </StyledPosterWrapper>
            <div>
                <StyledTitle>{ Title }</StyledTitle>
                <p>{ Plot }</p>
            </div>
        </StyledMovieSummary>
    );
}

const StyledMovieSummary = styled.div`
    color: ${colorPallete.light[20]};
    border: ${rem(1)} solid ${colorPallete.light[96]};
    display: flex;
    flex-direction: column;
    text-decoration: none;
    transition: all .5s;

    &:hover {
        background: ${colorPallete.light[98]};
        box-shadow: 0 ${rem(10)} ${rem(15)} ${rem(-3)} rgba(0, 0, 0, 0.1), 0 ${rem(4)} ${rem(6)} ${rem(-2)} rgba(0, 0, 0, 0.05);
    }

    }
    svg {
        color: ${colorPallete.light[80]};
    }
`;
const StyledPosterWrapper = styled.div`
    max-height: ${rem(370)};
    overflow: hidden;
`;

const StyledSmallImage = styled.img`
    height: auto;
    width: 100%;
    transition: opacity .5s;

    &:hover {
        opacity: .8;
    }

`;

const StyledTitle = styled.h2`
    font-size: ${rem(16)};
    font-weight: 600;
    padding: ${rem(16)} ${rem(8)};
`;

const textBase = css`
  font-size: 14px;
  text-decoration: none;
`;

export default MovieSummary;