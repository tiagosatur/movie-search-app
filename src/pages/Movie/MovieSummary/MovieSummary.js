import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faFileImage } from '@fortawesome/free-regular-svg-icons'
import styled from 'styled-components';
import { colorPallete, rem } from '../../../style'

const MovieSummary = props => {
    const {
        data: {
            Title,
            Plot,
            Poster
        },
    } = props;

    return (
        <StyledMovieSummary>
            <StyledPosterWrapper>
                {Poster !== 'N/A' 
                    ? <StyledSmallImage src={Poster} alt={Title} /> 
                    : <SVGWrapper>
                        <FontAwesomeIcon icon={faFileImage} style={{fontSize: '100px'}} />
                    </SVGWrapper>
                }
            </StyledPosterWrapper>
            <div>
                <StyledTitle>{ Title }</StyledTitle>
                <StyledPlot>{ Plot }</StyledPlot>
            </div>
        </StyledMovieSummary>
    );
}

const StyledMovieSummary = styled.div`
    color: ${colorPallete.light[20]};
    border: ${rem(1)} solid ${colorPallete.light[96]};
    display: flex;
    flex-direction: column;
    text-decoration: none;
    transition: all .5s;
    &:hover {
        background: ${colorPallete.light[98]};
        box-shadow: 0 ${rem(10)} ${rem(15)} ${rem(-3)} rgba(0, 0, 0, 0.1), 0 ${rem(4)} ${rem(6)} ${rem(-2)} rgba(0, 0, 0, 0.05);
    }
    }
    svg {
        color: ${colorPallete.light[80]};
    }
`;
const StyledPosterWrapper = styled.div`
    max-height: ${rem(370)};
    overflow: hidden;
`;

const StyledSmallImage = styled.img`
    height: auto;
    width: 100%;
    transition: opacity .5s;
    &:hover {
        opacity: .8;
    }
`;

const StyledTitle = styled.h2`
    color: ${colorPallete.light[13]};
    font-size: ${rem(16)};
    font-weight: 200;
    padding: ${rem(16)} ${rem(8)};
`;

const StyledPlot = styled.p`
  font-size: ${rem(14)};
  &:hover {
    color: ${colorPallete.primary};
  }
`;

const SVGWrapper = styled.div`
    align-items: center;
    display: flex;
    justify-content: center;
    padding: ${rem(16)};
`;



export default MovieSummary;