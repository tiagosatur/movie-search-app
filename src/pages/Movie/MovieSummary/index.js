export { default as MovieSummary } from './MovieSummary';