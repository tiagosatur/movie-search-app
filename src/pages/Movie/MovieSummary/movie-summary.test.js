import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MovieSummary from './MovieSummary';

describe('Movie Summary component test', () => {
    const 
        Title = 'Avatar',
        Plot = 'Some plot writen ramdomly',
        Poster = 'http://www.mayimage.com/#123'
    
    
    const wrapper =  shallow(
        <MovieSummary data={{Title, Poster, Plot}} />
    );

    it('should call App snapshot correctly', () => {
        const tree = renderer
            .create(wrapper)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})


