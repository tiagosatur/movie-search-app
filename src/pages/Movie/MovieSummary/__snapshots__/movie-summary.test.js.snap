// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Summary component test should call App snapshot correctly 1`] = `
<div
  className="sc-bdVaJa bUujvC"
>
  <div
    className="sc-bwzfXH laJbGY"
  >
    <img
      alt="Avatar"
      className="sc-htpNat jPUyTb"
      src="http://www.mayimage.com/#123"
    />
  </div>
  <div>
    <h2
      className="sc-bxivhb bvvINP"
    >
      Avatar
    </h2>
    <p
      className="sc-ifAKCX cRvWwb"
    >
      Some plot writen ramdomly
    </p>
  </div>
</div>
`;
